setup/blocks/base_block.php

```diff
@@ -1,28 +1,53 @@
 <?php
 
-class BaseBlock {
-    public function __construct($block_name = '', $block_params = array(), $group_params = array(), $fields_params = array()){
-
-        if (empty($block_name) || empty($block_params) || empty($group_params) || empty($fields_params)) {
-            return;
-        }
+abstract class BaseBlock {
+    public function __construct(){
+        $this->validate_required_constants();
+        $this->register();
+    }
 
+    protected function register(){
         if( function_exists('create_acf_block') ) {
-            create_acf_block($block_name, $block_params);
+            create_acf_block($this->name(), $this->params());
         }
 
         if( function_exists('acf_add_local_field_group') ) {
-            acf_add_local_field_group($group_params);
+            acf_add_local_field_group($this->groupParams());
         }
 
         if( function_exists('acf_add_local_field') ) {
-            acf_add_local_field($fields_params);
+            acf_add_local_field($this->fieldsParams());
+        }
+    }
+
+    protected function validate_required_constants() {
```

Comment:
Voglio aiutare chi eredita da questa classe ad implementare la classe che sta estendendo in modo corretto. `get_called_class()` restituisce il nome della classe chiamata; se non ha le costanti così come le hai pensate tu, ci rompiamo e diamo un messaggio specifico.